src/controllers/category.js

```diff
@@ -1,15 +1,48 @@
-/* eslint-disable import/extensions */
-const logger = require("../middleware/logger");
+const { redisClient } = require("../middleware/redisClient");
 const Category = require("../models/category");
 const asyncHandler = require("../middleware/async");
-const ErrorResponse = require("../utils/error-response");
+
+const cacheKey = "CATEGORY";
 
 const getCategories = asyncHandler(async (req, res, next) => {
   try {
```

Comment:
The implementation of `asyncHandler()` and including the `try...catch` idiom is wrong as it's repetitive. If you go the `asyncHandler` route, don't use `try...catch` and vice versa.

Reply:
ok i'll resove this